src/devices/LidarLiteV3/LidarLiteV3.cs

```diff
@@ -0,0 +1,358 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading.Tasks;
+using System.Buffers.Binary;
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.Threading;
+using Iot.Device.DistanceSensor.Models.LidarLiteV3;
+
+namespace Iot.Device.DistanceSensor
+{
+    /// <summary>
+    /// Lidar Lite v3 is a long-range fixed position distance sensor by Garmin.
+    /// </summary>
+    public class LidarLiteV3 : IDisposable
+    {
+        /// <summary>
+        /// Default address for LidarLiteV3
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x62;
+
+        private GpioController _gpioController;
+        private I2cDevice _i2cDevice;
+        private int _powerEnablePin;
+
+        /// <summary>
+        /// Initialize the LidarLiteV3
+        /// </summary>
+        /// <param name="i2cDevice">I2C device</param>
+        /// <param name="gpioController">GPIO controller</param>
+        /// <param name="powerEnablePin">The pin number used to control power to the device</param>
+        public LidarLiteV3(I2cDevice i2cDevice, GpioController? gpioController = null, int powerEnablePin = -1)
+        {
+            _gpioController = gpioController ?? new GpioController();
+            _i2cDevice = i2cDevice;
+            _powerEnablePin = powerEnablePin;
+
+            if (_powerEnablePin != -1)
+            {
+                _gpioController.OpenPin(_powerEnablePin, PinMode.Output);
+                PowerOn();
+            }
+
+            Reset();
+        }
+
+        /// <summary>
+        /// Power off the device if GPIO controller and power enable pin is provided.
+        /// </summary>
+        public void PowerOff()
+        {
+            if (_powerEnablePin != -1)
+            {
+                _gpioController.Write(_powerEnablePin, PinValue.Low);
+            }
+            else
+            {
+                throw new InvalidOperationException("Cannot power off without providing GPIO controller and power enable pin.");
+            }
+        }
+
+        /// <summary>
+        /// Power on the device if GPIO controller and power enable pin is provided.
+        /// </summary>
+        public void PowerOn()
+        {
+            if (_powerEnablePin != -1)
+            {
+                _gpioController.Write(_powerEnablePin, PinValue.High);
+            }
+            else
+            {
+                throw new InvalidOperationException("Cannot power off without providing GPIO controller and power enable pin.");
+            }
+        }
+
+        /// <summary>
+        /// Reset FPGA, all registers return to default values
+        /// </summary>
+        public void Reset()
+        {
+            WriteRegister(Register.ACQ_COMMAND, 0x00);
+        }
+
+        /// <summary>
+        /// Measure distance in cm
```

Comment:
Won't comment on all of these, but all of the methods you have that are returning an int in centimeters, should instead be returning `UnitsNet.Length` types instead so that consumer can decide if they want it on cm or some other unit instead.

Reply:
Updated to use UnitsNet.